--- a/kte-plugin/ktpintegration/CMakeLists.txt
+++ b/kte-plugin/ktpintegration/CMakeLists.txt
@@ -24,8 +24,13 @@
                       ${KTP_MODELS_LIBRARIES}
                       ${KTP_WIDGETS_LIBRARIES}
 )
+set_target_properties(inftube PROPERTIES
+                              INSTALL_RPATH_USE_LINK_PATH TRUE
+                              SKIP_BUILD_PATH TRUE
+                              BUILD_WITH_INSTALL_RPATH TRUE
+                              INSTALL_RPATH ${PRIVATE_LIB_INSTALL_DIR})
 
-install(TARGETS inftube ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS inftube DESTINATION ${PRIVATE_LIB_INSTALL_DIR})
 
 # the tube handlers
 kde4_add_executable(infinoteclienttubehandler clienttubehandler.cpp)
--- a/common/CMakeLists.txt
+++ b/common/CMakeLists.txt
@@ -36,5 +36,10 @@
     ${KDE4_KDNSSD_LIBS}
     ${LIBQINFINITY_LIBRARIES}
 )
+set_target_properties(inftube PROPERTIES
+                              INSTALL_RPATH_USE_LINK_PATH TRUE
+                              SKIP_BUILD_PATH TRUE
+                              BUILD_WITH_INSTALL_RPATH TRUE
+                              INSTALL_RPATH ${PRIVATE_LIB_INSTALL_DIR})
 
-install(TARGETS kobbycommon DESTINATION ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kobbycommon DESTINATION ${PRIVATE_LIB_INSTALL_DIR})
--- a/kioslave/CMakeLists.txt
+++ b/kioslave/CMakeLists.txt
@@ -20,6 +20,8 @@
     ${KDE4_KTEXTEDITOR_LIBS}
     ${KDE4_KDNSSD_LIBS}
 )
+set_target_properties(kio_infinity PROPERTIES
+                                   INSTALL_RPATH ${PRIVATE_LIB_INSTALL_DIR})
 
 install( TARGETS kio_infinity DESTINATION ${PLUGIN_INSTALL_DIR} )
 install( FILES infinity.protocol DESTINATION  ${SERVICES_INSTALL_DIR} )
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,6 +9,9 @@
 find_package(KDE4 REQUIRED)
 find_package(Libqinfinity REQUIRED)
 
+SET( PRIVATE_LIB_INSTALL_DIR lib/ktecollaborative )
+SET( CMAKE_INSTALL_RPATH ${PRIVATE_LIB_INSTALL_DIR} )
+
 # TODO: make optional?
 set(IS_KTP_INTERNAL_MODULE TRUE) # aw yeah
 find_package(TelepathyQt4 0.8.9 REQUIRED)
